### PR TITLE
THRIFT-5649: add github action for go

### PR DESCRIPTION
**Pull request by Jimexist**
add github action for go

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [x] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

**Comment by Jimexist** [comment]
the cross test failure is irrelevant to the changes in this pr

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -51,6 +51,80 @@ jobs:
...
 
+  lib-go:
+    needs: compiler
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '>=1.17.0'
```
Comment (fishy) [inline .github/workflows/build.yml:62]: the current supported go versions are 1.18.x and 1.19.x: https://github.com/apache/thrift/blob/master/LANGUAGES.md, not 1.17.x.

but looks like it actually picks the latest version (1.19.1) from the logs, so I think we probably actually need two actions, one for `lib-go-1.18` with `<1.19.0` and one for `lib-go-1.19` with `>=1.19.0`, or something like that.

Comment (Jimexist) [reply .github/workflows/build.yml:62]: thanks for the suggestion - will setup a new jira to follow up

Comment (Jimexist) [reply .github/workflows/build.yml:62]: see https://issues.apache.org/jira/browse/THRIFT-5651

**Pull request merged**